lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java

```diff
@@ -205,6 +197,86 @@ public APIGatewayProxyResponseEvent handleRequest(
         }
     }
 
+    private APIGatewayProxyResponseEvent getUserIdentityEndpointResponse(
+            IpvSessionItem ipvSessionItem,
+            String userId,
+            ClientOAuthSessionItem clientOAuthSessionItem)
+            throws CiRetrievalException, VerifiableCredentialException, SqsException,
+                    HttpResponseExceptionWithErrorBody, CredentialParseException,
+                    UnrecognisedCiException {
+
+        var ipvSessionId = ipvSessionItem.getIpvSessionId();
+
+        AuditEventUser auditEventUser =
+                new AuditEventUser(
+                        userId,
+                        ipvSessionId,
+                        clientOAuthSessionItem.getGovukSigninJourneyId(),
+                        null);
+
+        var contraIndicatorsVc =
+                ciMitService.getContraIndicatorsVc(
+                        userId, clientOAuthSessionItem.getGovukSigninJourneyId(), null);
+
+        var contraIndicators = ciMitService.getContraIndicators(contraIndicatorsVc);
+
+        var vcs = sessionCredentialsService.getCredentials(ipvSessionId, userId);
+
+        UserIdentity userIdentity =
+                userIdentityService.generateUserIdentity(
+                        vcs, userId, ipvSessionItem.getVot(), contraIndicators);
+        userIdentity.getVcs().add(contraIndicatorsVc.getVcString());
+        if (configService.enabled(TICF_CRI_BETA)
+                && (ipvSessionItem.getRiskAssessmentCredential() != null)) {
+            userIdentity.getVcs().add(ipvSessionItem.getRiskAssessmentCredential());
+        }
+
+        sendIdentityIssuedAuditEvent(
+                ipvSessionItem, auditEventUser, contraIndicators, userIdentity);
+
+        ipvSessionService.revokeAccessToken(ipvSessionItem);
+
+        deleteSessionCredentials(ipvSessionId);
+
+        var message =
+                new StringMapMessage()
+                        .with(
+                                LOG_LAMBDA_RESULT.getFieldName(),
+                                "Successfully generated user identity response.")
+                        .with(LOG_VOT.getFieldName(), ipvSessionItem.getVot());
+        LOGGER.info(message);
+
+        return ApiGatewayResponseGenerator.proxyJsonResponse(HTTPResponse.SC_OK, userIdentity);
+    }
+
+    private APIGatewayProxyResponseEvent getReverificationEndpointResponse(
+            IpvSessionItem ipvSessionItem, String userId) {
+
+        var ipvSessionId = ipvSessionItem.getIpvSessionId();
+
+        if (ipvSessionItem.getVot().equals(Vot.P2)) {
+            ipvSessionService.revokeAccessToken(ipvSessionItem);
+            deleteSessionCredentials(ipvSessionId);
+
+            ReverificationResponse successResponse = ReverificationResponse.successResponse(userId);
+
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HTTPResponse.SC_OK, successResponse);
+        } else {
+            ipvSessionService.revokeAccessToken(ipvSessionItem);
```

Comment:
The first two lines of both branches are identical, so we could pull them out.

However, I think it would be even better to move this to the end of the handler method instead (since it also happens in the user-identity endpoint) - we should invalidate the access token only once we're confident we'll return a proper response (i.e. 200).

I don't think the rest of this method can actually fail, but it would convey intent more clearly :)